src/Microsoft.Identity.Abstractions/DownstreamApi/IAuthorizationHeaderProvider_T.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Abstractions
+{
+    /// <summary>
+    /// <inheritdoc cref="IAuthorizationHeaderProvider"/>
+    /// </summary>
+    /// <typeparam name="TResult">The result type.</typeparam>
+    public interface IAuthorizationHeaderProvider<TResult>
```

Comment:
Do we need to add matching ForUser and ForApp methods like non-generic interface version?

Reply:
I don't think so. This is done through the options, RequestAppToken. The specifics for user and app are older APIs